pkg/protocol/engine/ledger/ledger/ledger.go

```diff
@@ -610,14 +611,13 @@ func (l *Ledger) processStateDiffTransactions(stateDiff mempool.StateDiff) (spen
 		{
 			// input side
 			for _, inputRef := range inputRefs {
-				inputState, outputErr := l.Output(inputRef.OutputID())
-				if outputErr != nil {
+				stateWithMetadata, stateError := l.memPool.StateMetadata(inputRef)
```

Comment:
why do we use the mempool directly here now?

Reply:
because we are applying the state from the mempool actually